controllers/overlaynetwork_controller.go

```diff
@@ -61,7 +65,14 @@ func (r *OverlayNetworkReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, fmt.Errorf("failed to get OverlayNetwork %q: %v", req.NamespacedName, err)
 	}
 
-	netAttachDef := renderNetAttachDef(overlayNetwork)
+	if overlayNetwork.Spec.Name == "" {
```

Comment:
In the CRD definition the field is not optional, so I believe it cannot be empty.